File: 1828-queries-on-number-of-points-inside-a-circle/1828-queries-on-number-of-points-inside-a-circle.py
class Solution:
    def countPoints(self, points: List[List[int]], queries: List[List[int]]) -> List[int]:
        
        answer = []

        def inside(px, py, x, y, r):

            dx = px - x
            dy = py - y
            return dx * dx + dy * dy <= r * r
            
        for q in queries:
            x, y, r = q
            
            count = 0

            for p in points:
                px, py = p
                if inside(px, py, x, y, r):
                    count += 1
            
            answer.append(count)
        
        return answer

File: 1828-queries-on-number-of-points-inside-a-circle/1828-queries-on-number-of-points-inside-a-circle.cpp
class Solution {
    bool inside(int px, int py, int x, int y, int r) {
            int dx = px - x;
            int dy = py - y;

            return dx * dx + dy * dy <= r * r;
        }
public:
    vector<int> countPoints(vector<vector<int>>& points, vector<vector<int>>& queries) {
        vector<int> answer;

        

        for (auto &q : queries) {
            int x = q[0];
            int y = q[1];
            int r = q[2];

            int cnt = 0;
            for (auto &p : points) {
                if (inside(p[0], p[1], x, y, r)) {
                    cnt++;
                }
            }
            answer.push_back(cnt);
        }
        return answer;
    }
};